f/frizzle/comapeo/comapeo_observations.py

```diff
@@ -37,24 +39,40 @@ def main(
     comapeo_projects = fetch_comapeo_projects(
         comapeo_server_base_url, comapeo_access_token, comapeo_project_blocklist
     )
+
+    # Run culminates in success if there were no projects returned by the API
+    if len(comapeo_projects) == 0:
+        logger.info(
+            "No projects fetched. Skipping data processing and database writing."
+        )
+        return
+
     logger.info(f"Fetched {len(comapeo_projects)} projects.")
 
-    comapeo_data = process_comapeo_data(
-        comapeo_server_base_url, comapeo_access_token, comapeo_projects, attachment_root
+    attachment_failed = False
+
+    comapeo_data, attachment_failed = process_comapeo_data(
+        comapeo_server_base_url,
+        comapeo_access_token,
+        comapeo_projects,
+        db_table_prefix,
+        attachment_root,
+        attachment_failed,
```

Comment:
I see no reason that `attachment_failed` has to be sent from outside.  It returns the boolean and that should be enough.